website/docs/externaldata.md

```diff
@@ -168,4 +129,186 @@ type Item struct {
+
+> 📎 To avoid multiple calls to the same provider, recommendation is to batch the keys list to send a single request.
+
+Example template:
+https://github.com/open-policy-agent/gatekeeper/blob/master/test/externaldata/dummy-provider/policy/template.yaml
+
+## External data for Gatekeeper mutating webhook
+
+External data can be used in conjunction with [Gatekeeper mutating webhook](mutation.md).
+
+### API
+
+You can specify the details of the external data provider in the `spec.parameters.assign.externalData` field of `AssignMetadata` and `Assign`.
+
+> Note: `spec.parameters.assign.externalData`, `spec.parameters.assign.value` and `spec.parameters.assign.fromMetadata` are mutually exclusive.
+
+| Field                            | Description                                                                                                                                                 |
+|----------------------------------|-------------------------------------------------------------------------------------------------------------------------------------------------------------|
+| `provider`<br/>String             | The name of the external data provider.                                                                                                                     |
+| `dataSource`<br/>DataSource       | Specifies where to extract the data that will be sent to the external data provider.<br/>- `ValueAtLocation` (default): extracts an array of values from the path that will be modified. See [mutation intent](mutation.md#intent) for more details.<br/>- `Username`: The name of the Kubernetes user who initiated the admission request.  |
+| `failurePolicy`<br/>FailurePolicy | The policy to apply when the external data provider returns an error.<br/>- `UseDefault`: use the default value specified in `spec.parameters.assign.externalData.default`<br/>- `Ignore`: ignore the error and do not perform any mutations.<br/>- `Fail` (default): do not perform any mutations and return the error to the user.                                       |
+| `default`<br/>String              | The default value to use when the external data provider returns an error and the failure policy is set to `UseDefault`.                                    |
+
+### `AssignMetadata`
+
+```yaml
+apiVersion: mutations.gatekeeper.sh/v1beta1
+kind: AssignMetadata
+metadata:
+  name: annotate-owner
+spec:
+  match:
+    scope: Namespaced
+    kinds:
+    - apiGroups: ["*"]
+      kinds: ["Pod"]
+  location: "metadata.annotations.owner"
+  parameters:
+    assign:
+      externalData:
+      provider: my-provider
+      dataSource: Username
+```
+
+<details>
+<summary>Provider response</summary>
+
+```json
+{
+  "apiVersion": "externaldata.gatekeeper.sh/v1alpha1",
+  "kind": "ProviderResponse",
+  "response": {
+    "idempotent": true,
+    "items": [
+      {
+        "key": "kubernetes-admin",
```

Comment:
better example would be an email maybe and it can return a fake org/team name or something?